### ISSUE TITLE
Dynamically changable effects

### ISSUE THREAD

**Issue by Zomis** [Cardshifter/Cardshifter#216]
Consider for example a card, let's call it "Megaman" that has: "At the end of your turn, deal 2 damage to all opponent creatures"

And then a card that says: "Your Megaman's deal 4 damage to all opponent creatures instead."

The key here is that the effect of the Megaman should be changed *without changing the code for the Megaman*. i.e. no if-else statement to check if the other card is in play or not.

**Comment by Phrancis** [comment]
Perhaps I'm reading this wrong but would this not be principally a matter of being able to have an effect applicable on a specific card? (rather than a card type or zone)? Like, for instance (notice new keyword in filter)

    name: "gives Megaman +2 attack",
    afterPlay: [
        {   
            res: ATTACK,
            priority: 1,
            change: 2,
            filter: { specificCreature: "Megaman", owner: "owner" }
        },

**Comment by Zomis** [comment]
No, this does not change the *attack* of Megaman, it changes one of the *effects* on Megaman.
